solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/test/java/com/solace/spring/cloud/stream/binder/util/XMLMessageMapperTest.java

```diff
@@ -484,6 +485,149 @@ public void testMapConsumerErrorSpringMessageToXMLMessage_WithProperties() {
 		assertEquals(consumerProperties.getErrorMsgTtl().longValue(), xmlMessage.getTimeToLive());
 	}
 
+	@Test
+	public void testMapProducerSpringMessageToXMLMessage_WithExcludedHeader() throws SDTException {
+		String testPayload = "testPayload";
+		List<String> excludedHeaders = new ArrayList<>(Arrays
+				.asList("io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor.SCOPE"));
+		Message<?> testSpringMessage = new DefaultMessageBuilderFactory().withPayload(testPayload)
+				.setHeader(
+						"io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor.SCOPE",
+						"any")
+				.build();
+
+		XMLMessage xmlMessage = xmlMessageMapper.map(testSpringMessage, excludedHeaders);
+		Mockito.verify(xmlMessageMapper).map(testSpringMessage, excludedHeaders);
+
+		assertEquals(null, xmlMessage.getProperties()
+				.getMap("io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor.SCOPE"));
+	}
+
+	@Test
+	public void testMapProducerSpringMessageToXMLMessage_WithExcludedHeader_ShouldNotMatchPartially() throws SDTException {
+		String testPayload = "testPayload";
+		List<String> excludedHeaders = new ArrayList<>(Arrays
+				.asList("io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor.SCOPE"));
+		Message<?> testSpringMessage = new DefaultMessageBuilderFactory().withPayload(testPayload)
+				.setHeader(
+						"io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor",
+						"any")
+				.build();
+
+		XMLMessage xmlMessage = xmlMessageMapper.map(testSpringMessage, excludedHeaders);
+		Mockito.verify(xmlMessageMapper).map(testSpringMessage, excludedHeaders);
+		assertEquals("any", xmlMessage.getProperties()
+				.get("io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor"));
+	}
+
+	@Test
+	public void testMapProducerSpringMessageToXMLMessage_WithExcludedHeader_ShouldNotFilterSolaceHeader() throws SDTException {
+		String testPayload = "testPayload";
+		MessageBuilder<?> messageBuilder = new DefaultMessageBuilderFactory()
+				.withPayload(testPayload)
```

Comment:
change this line to: `.withPayload(new SerializableFoo("a", "b"))`

And add header: `.setHeader("test-serializable-header", new SerializableFoo("a", "b"))`